File: test_shopee_last_try/4b.py
from attr import has
import pandas as pd

df1 = pd.read_csv('Analisis excel/test_shopee_last_try/Orders.csv')
df1['Order Date'] = pd.to_datetime(df1['Order Date'])
df2 = pd.read_csv('Analisis excel/test_shopee_last_try/Customer.csv')
df = pd.merge(df1, df2, on='Customer ID', how='inner')
print(df)
filter = df1.groupby('Customer ID')['Order Date'].min().reset_index(name='tanggal_beli_pertama')
hasil_filter = df[df['Order Date'].isin(filter['tanggal_beli_pertama'])]

print()
hasil = hasil_filter.groupby(['City', 'Order Date'])['Customer ID'].count().reset_index(name='total_order').sort_values('total_order', ascending=False).reset_index(drop=True)
print(hasil)
hasil = hasil_filter.groupby(['City', 'Order Date'])['Customer ID'].nunique().reset_index(name='jumlah').sort_values('jumlah', ascending=False).reset_index(drop=True)
print(hasil)

File: test_shopee_last_try/4b.r
library(dplyr)
library(tidyr)

customer <- read.csv("Analisis excel/Youtube/5 soal/Customer.csv")
print(head(customer))

order <- read.csv("Analisis excel/Youtube/5 soal/Orders.csv")
order$Order.Date <- as.Date(order$Order.Date, "%m/%d/%Y")
print(head(order))

# Inner join customer and order data frames
df <- inner_join(customer, order, by = "Customer.ID", relationship ="many-to-many")
filter <- order %>% group_by(Customer.ID) %>% summarise(first_order = min(Order.Date))

df <- df[df$Order.Date %in% filter$first_order,]
print(df)

hasil <- df %>% group_by(City, Order.Date) %>% summarise(total_pembeli = n()) %>% arrange(desc(total_pembeli))
print(hasil)
print(dim(hasil))
hasil2 <- df %>% group_by(City, Order.Date) %>% summarise(total_pembeli = n_distinct(Customer.ID)) %>% arrange(desc(total_pembeli))
print(hasil2, n=20)
print(dim(hasil2))